### ISSUE TITLE
Assertion failed error

### ISSUE THREAD

**Issue by bchr02** [johntitus/node-horseman#20]
I keep getting the following error when using Internet Explorer
>Assertion failed: !current_buffer, file src\node_http_parser.cc, line 387

I have a Express website that uses AJAX request to pull data in using node-horseman. Whenever Internet Explorer is used I get the above error. Google Chrome however does not have any issues.

I am using node v0.10.36

While troubleshooting, I have even removed everything but the ````var x = new horseman```` initialization and I still get the error. As soon as I remove that, the issue goes away.

Please help. Thanks.

**Comment by bchr02** [comment]
IE is v11.

**Comment by bchr02** [comment]
I created a bare bones example that shows the issue. Strange thing is, while using this code, the issue is also occurring on Chrome.

>var express = require('express');
var app = express();
var Horseman = require('node-horseman');

>app.get('/', function (req, res) {
	var horseman = new Horseman();
	horseman.close();
	res.send('Hello World!');
})

>var server = app.listen(3333, function () {
	console.log('listening at port: ', server.address().port);
});

**Comment by bchr02** [comment]
@johntitus Just curious when do you think you might have a chance to take a look at this? Thanks.

**Comment by johntitus** [comment]
Sorry, been distracted. Looks very similar to this:
https://github.com/johntitus/node-horseman/issues/15

Though I don't know the cause of it, they were able to resolve it by using a newer version of express. What version do you have?

**Comment by bchr02** [comment]
I do have the latest version of express.

I am using node v0.10.36. I cannot use a newer version of node yet because of another module incompatibility but I did try the test code on v0.12 and it worked without issues.

Can you help me to get it working with v0.10.36 so I don't have to use another module? Thanks.

**Comment by johntitus** [comment]
I tried to figure it out in that previous issue with no luck. He did have success by spawning a child process to run horseman. Maybe give that a try?

**Comment by bchr02** [comment]
>He did have success by spawning a child process to run horseman.

I read through the post and didn't see where that was stated. Maybe I missed it. Can you explain how I would do this? Are you suggesting having two versions of node running side by side?

**Comment by johntitus** [comment]
I don't have a lot of info, sorry. I assume he used the `spawn`

https://nodejs.org/api/child_process.html


Me (in an email conversation)
In the interim, you may want to try spawning a child process to run horseman.

Him (in response)
Thanks a lot for quick turn around John.I ll right away create it as a  issue in node-horseman.And thanks for the interim solution.We are following child process approach right now and its working fine.

**Comment by viveksrinivasanss** [comment]
hi bchr02 ...Sorry for the late response.U might have figured out a way to do it by this time.

this is how i did it
```
execFile('node',["your path"],{maxBuffer: 5000*1024},function(err,stdout,stderr){
        if(err) {
            console.log(err);
        }
        console.log(,stdout);
    });	
```

But the assertion problem still exist for me when I dont trigger it as a child process.

**Comment by viveksrinivasanss** [comment]
If you are using child process as a solution. U might have come across this .When I execute child process it started consoling these junk console statements to the screen.

**phantom stdout: 'null' is not an object (evaluating 'segmentCookie.replace')**
**phantom stdout:   http://www1.macys.com/shop/product/michael-michael-kors-flat-front-dress-pants?ID=1166625&CategoryID=89:402**
**phantom stdout: 'null' is not an object (evaluating 'a.appendChild')**
**phantom stdout:**   **http://assets.macys.com/navapp/web20/assets/combo/catalog.pdp.responsive.member.responsive**  **script-min-13.js?timenow=201503180458:2**


Is there a  way to avoid these console statements??
somewhere when I create horseman instance can i mention that??

**Comment by bchr02** [comment]
@viveksrinivasanss No, I haven't fixed yet, I have been busy with other things, so your example will be most helpful and save me time. Thank you very much.

Once I give it a try, I'll let you know if I come across those junk console messages, but it might not be for several weeks because I will be travelling.

**Comment by bchr02** [comment]
Hopefully @johntitus might have some answers for you. I got to be honest though, I switched to this repo from [nightmare](https://github.com/segmentio/nightmare) and now I am thinking of switching back because of this "Assertion" issue. I was kinda hoping that a fix would be found instead of this workaround.

**Comment by johntitus** [comment]
Linking to #40 as I suspect it's based on the blocking nature of Horseman.

**Comment by pitaj** [comment]
Actually, this issue is caused by executing the code in the context of `server.create`

You can use process.nextTick to fix the issue as shown here: [joyent/node#1574](/joyent/node/issues/1574)

**Comment by johntitus** [comment]
pitaj, thanks for the tip. I tested the original code provided by @bchr02 and it seems to work fine now, unmodified. I'm using express 4.13.1 and node 0.12.7 and the latest horseman. 

Marking code as somehow this doesn't seem to be an issue anymore.
